nucleus/annotation_uploader.py

```diff
@@ -0,0 +1,198 @@
+import json
+from typing import TYPE_CHECKING, Iterable, List, Sequence
+
+from nucleus.annotation import Annotation, SegmentationAnnotation
+from nucleus.async_utils import (
+    FileFormField,
+    FormDataContextHandler,
+    make_many_form_data_requests_concurrently,
+)
+from nucleus.constants import MASK_TYPE, SERIALIZED_REQUEST_KEY
+from nucleus.payload_constructor import (
+    construct_annotation_payload,
+    construct_segmentation_payload,
+)
+
+if TYPE_CHECKING:
+    from . import NucleusClient
+
+
+def accumulate_dict_values(dicts: Iterable[dict]):
+    """
+    Accumulate a list of dicts into a single dict using summation.
+    """
+    result = {}
+    for d in dicts:
+        for key, value in d.items():
+            if key not in result or key == "dataset_id":
+                result[key] = value
+            else:
+                result[key] += value
+    return result
+
+
+class AnnotationUploader:
+    def __init__(self, dataset_id: str, client: "NucleusClient"):  # noqa: F821
+        self.dataset_id = dataset_id
+        self._client = client
+
+    def upload(
+        self,
+        annotations: Iterable[Annotation],
+        batch_size: int = 5000,
+        update: bool = False,
+        remote_files_per_upload_request: int = 20,
+        local_files_per_upload_request: int = 10,
+        local_file_upload_concurrency: int = 30,
+    ):
```

Comment:
Here we need a docstring, I would specifically look for reasoning behind the default magic numbers. How and why would you choose to use something other than the default?

Edit: Saw the public facing documentation, would be great to have it here as well!

Reply:
I'm not a huge fan of duplicating the docs in case they get updated there and not here. The main place should be in the user facing classes, this is an internal class. Added some docstring saying to see dataset.annotate for documentation.